### PR TITLE
Add a EdgeInsetsDirectional.all constructor similar to EdgeInsets.all

### PR DESCRIPTION
**Pull request by allysonjp715**
Adding a convenience constructor for EdgeInsetsDirectional that EdgeInsets already has.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/painting/edge_insets.dart [hunk 1]
```diff
@@ -663,6 +663,22 @@ class EdgeInsetsDirectional extends EdgeInsetsGeometry {
...
+  /// {@tool snippet}
+  ///
+  /// Typical eight-pixel margin on all sides:
+  ///
+  /// ```dart
+  /// const EdgeInsetsDirectional.all(8.0)
+  /// ```
+  /// {@end-tool}
+  const EdgeInsetsDirectional.all(double value)
+      : start = value,
```
Comment (xu-baolin) [inline packages/flutter/lib/src/painting/edge_insets.dart:677]: Indent 2 spaces

**Review by xu-baolin — APPROVED** [review]
LGTM with a small nit.

File: packages/flutter/lib/src/painting/edge_insets.dart (hunk 1 above)
Comment (allysonjp715) [reply packages/flutter/lib/src/painting/edge_insets.dart]: Done.

Comment (xu-baolin) [reply packages/flutter/lib/src/painting/edge_insets.dart]: Thanks for your contribution!

**Comment by patrikheinonen** [comment]
Hello @allysonjp715 @xu-baolin 

Why was this mr accepted into the codebase. How is this convenient. Also note that EdgeInsets.all is cheaper to resolve than EdgeInsetsDirectional.all and if this mr is not reverted, at least that fact should be added to the documentation of the method.

**Pull request merged**